kafka-scheduler/src/main/java/com/mesosphere/dcos/kafka/offer/PersistentOfferRequirementProvider.java

```diff
@@ -20,475 +19,462 @@
-            .addResources(ResourceUtils.getDesiredRanges(
-                    role,
-                    principal,
-                    "ports",
-                    Arrays.asList(
-                            Range.newBuilder()
-                                    .setBegin(port)
-                                    .setEnd(port).build())));
-
-    if (brokerConfig.getDiskType().equals("MOUNT")) {
-      taskBuilder.addResources(ResourceUtils.getDesiredMountVolume(
-            role,
-            principal,
-            brokerConfig.getDisk(),
-            containerPath));
-    } else {
-      taskBuilder.addResources(ResourceUtils.getDesiredRootVolume(
-            role,
-            principal,
-            brokerConfig.getDisk(),
-            containerPath));
+    private CommandInfo getNewBrokerCmd(KafkaSchedulerConfiguration config, int brokerId, Long port, String containerPath)
+            throws ConfigStoreException {
+        String brokerCmd = getBrokerCmd(config);
+        String brokerName = OfferUtils.brokerIdToTaskName(brokerId);
+
+        Map<String, String> envMap = new HashMap<>();
+        envMap.put("TASK_TYPE", KafkaTask.BROKER.name());
+        envMap.put("FRAMEWORK_NAME", config.getServiceConfiguration().getName());
+        envMap.put("KAFKA_VER_NAME", config.getKafkaConfiguration().getKafkaVerName());
+        envMap.put("KAFKA_ZOOKEEPER_URI", config.getKafkaConfiguration().getKafkaZkUri());
+        envMap.put(KafkaEnvConfigUtils.toEnvName("zookeeper.connect"), config.getFullKafkaZookeeperPath());
+        envMap.put(KafkaEnvConfigUtils.toEnvName("broker.id"), Integer.toString(brokerId));
+        envMap.put(KafkaEnvConfigUtils.toEnvName("log.dirs"), containerPath + "/" + brokerName);
+        envMap.put(KafkaEnvConfigUtils.toEnvName("listeners"), "PLAINTEXT://:" + port);
+        envMap.put(KafkaEnvConfigUtils.toEnvName("port"), Long.toString(port));
+        envMap.put("KAFKA_HEAP_OPTS", getKafkaHeapOpts(config.getBrokerConfiguration().getHeap()));
+        return CommandInfo.newBuilder()
+                .setValue(brokerCmd)
+                .setEnvironment(OfferUtils.environment(envMap))
+                .build();
+    }
+
+    private CommandInfo getNewExecutorCmd(KafkaSchedulerConfiguration config, String configName, int brokerId)
+            throws ConfigStoreException {
+        BrokerConfiguration brokerConfiguration = config.getBrokerConfiguration();
+        ZookeeperConfiguration zookeeperConfiguration = config.getZookeeperConfig();
+        ExecutorConfiguration executorConfiguration = config.getExecutorConfiguration();
+        String frameworkName = config.getServiceConfiguration().getName();
+
+        final String executorCommand = "./executor/bin/kafka-executor server ./executor/conf/executor.yml";
+        Map<String, String> executorEnvMap = new HashMap<>();
+        executorEnvMap.put("JAVA_HOME", "jre1.8.0_91");
```

Comment:
`JAVA_HOME` is being set at multiple places, maybe we make this a config OR constant ?